pkg/kratos/service.go

```diff
@@ -504,16 +517,53 @@ func (s *Service) ParseRecoveryFlowMethodBody(r *http.Request) (*kClient.UpdateR
 }
 
 func (s *Service) ParseSettingsFlowMethodBody(r *http.Request) (*kClient.UpdateSettingsFlowBody, error) {
-	body := new(kClient.UpdateSettingsFlowWithPasswordMethod)
+	type MethodOnly struct {
+		Method string `json:"method"`
+	}
```

Comment:
small suggestion: this struct is only used here, we could define it once outside of the function, and make it start with a lowercase letter so it doesn't end up getting exported

Reply:
I think this field must be exported, because it has a json tag. If `method` is not capitalized then json package won't see its value